posts/2023-11-05-telegram-bot-for-file-sharing-using-quarkus.adoc

```diff
@@ -0,0 +1,495 @@
+
+Additionally, the entire source code for this project is available on https://github.com/iarmush/file-sharing-bot[GitHub].
+
+
+== Architecture and use case
+
+The primary user scenario unfolds as follows:
+
+1. The user initiates a conversation with a Telegram bot and sends an attachment.
+2. The Telegram bot forwards this message to our backend application.
+3. The backend application stores this attachment in MinIO storage.
+4. The user sends the command `/link` to the Telegram bot to obtain a shareable link.
+5. The backend application generates a link to our frontend application and provides it to the user.
+6. The user can copy and share this link, that accesses the frontend application which features a "Download" button.
+7. When the "Download" button is pressed, the frontend sends an HTTP request to our backend. The backend application retrieves all attachments associated with this user, compiles them into a zip file, and returns it to the user. Simultaneously, the backend application records the downloaded data in MongoDB.
+
+In terms of state diagrams:
+
+image::state_diagram.png[,600]
+
+
+== Backend
+For the backend, we'll be employing a conventional web server. Our journey begins with the https://quarkus.io/get-started/[Quarkus Get Started page].
+
+To get started, we need to follow step 1 on the provided link, which involves installing the Quarkus CLI.
+
+Now, we can execute:
+```bash
+quarkus --help
+```
+to get familiar with the main functionality. This command will be our main entry point for any application manipulation. With its assistance we can create a skeleton, in fact, it will have everything that we need, and it will even be a ready-to-deploy application.
+
+Before we create an application we have to choose which extensions (or dependencies) we’re going to use.
+
+We can find them on https://quarkus.io/extensions/ or by executing:
+```bash
+quarkus extension list
+```
+For our needs, it will be enough to add `quarkus-mongodb-panache`, `quarkus-resteasy-reactive` and `quarkus-quinoa`.
+
+Additionally, we have to manually add https://github.com/rubenlagus/TelegramBots - a library for creating telegram bots, and https://github.com/quarkiverse/quarkus-minio - MinIO java client by Quarkus.
+
+While in the root directory of our new project, run this command:
+```bash
+quarkus create app com.example:file-sharing-bot \
+--extension=quarkus-mongodb-panache,quarkus-resteasy,quarkus-quinoa \
+--gradle
+```
+
+We can already run our application with the default implementation. To do so, navigate to the root directory and execute the command to start our application:
+```bash
+quarkus dev
+```
+
+Now, you can open the recommended welcome page at http://localhost:8080/
```

Comment:
```suggestion
Now, you can open the welcome page at http://localhost:8080/
```
I think `recommended` it s bit weird and you already mention `default` several times after